solution/ui/regulations/eregs-vite/src/components/resources/SearchGovResults.vue

```diff
@@ -0,0 +1,208 @@
+const getHighlightTerms = (string, startChar, stopChar) => {
+    // pure fns for composition
+    const splitString = (str, splitChar) => str.split(splitChar);
+    const getFirstItem = (arr) => arr.shift();
+    const removeFirstItem = (arr) => {
+        getFirstItem(arr);
+        return arr;
+    };
+    const mapFn = (fn, arr, args = []) => arr.map((item) => fn(item, ...args));
+    const uniqArr = (arr) => Array.from(new Set(arr));
+
+    return uniqArr(
+        mapFn(
+            getFirstItem,
+            mapFn(
+                splitString,
+                removeFirstItem(splitString(string, startChar)),
+                [stopChar]
+            )
+        )
+    );
+};
+
+/**
+ * https://jsdoc.app/tags-param.html -- see "Documenting a destructuring parameter"
+ *
+ * @param supplementalContentItem {Object} - A piece of supplemental content
+ * @param supplementalContentItem.description {?string} - description of the piece of supplemental content
+ * @param supplementalContentItem.description_headline {?string} - alternate description of the piece of supplemental content.  Usually null
+ * @param supplementalContentItem.snippet {?string} - snippet of descriptive text containing special characters that are surrounding substrings to be emphasized with bold styles
+ * @param startChar {string} - character that indicates the start of a substring to highlight
+ * @param stopChar {string} - character that indicates the end of a substring to highlight
+ * @returns {string} - description with opening and closing span tags surrounding substrings that are to be emphasized
+ */
+const formatLinkTitle = (
+    { description, description_headline, snippet },
+    startChar,
+    stopChar
+) => {
+    let linkTitle = description_headline || description;
+
+    if (!snippet) {
+        return linkTitle;
+    }
+
+    const termsToHighlight = getHighlightTerms(snippet, startChar, stopChar);
+
+    termsToHighlight.forEach((term) => {
+        linkTitle = linkTitle.replaceAll(
+            term,
+            `<span class="${highlightClass}">${term}</span>`
+        );
+    });
+
+    return linkTitle;
+};
+
+/**
+ * @param snippet {?string} - snippet of descriptive text containing special characters that are surrounding substrings to be emphasized with bold styles
+ * @param startChar {string} - character that indicates the start of a substring to highlight
+ * @param stopChar {string} - character that indicates the end of a substring to highlight
+ * @returns {string} - snippet with special characters replaced with opening and closing span tags with a class used to add highlighting styles
+ */
+const formatSnippet = (snippet, startChar, stopChar) => {
+    if (!snippet) {
+        return snippet;
+    }
+
+    const reOpen = new RegExp(startChar, "g");
+    const reClose = new RegExp(stopChar, "g");
+
+    return snippet
+        .replace(reOpen, `<span class="${highlightClass}">`)
+        .replace(reClose, "</span>");
+};
+</script>
+
+<template>
```

Comment:
I know this is technically fine but we typically have template first on our files and then functions and whatnot.  Is this a Vue3 style change or is it just how you did it on this file.  We should be consistent.

Reply:
Going forward using the Vue 2.7/3 Composition API, the official docs have `<script>` above `<template>`.  This is in line with how React does it with `jsx`/`tsx` files.

https://vuejs.org/guide/extras/composition-api-faq.html#what-is-composition-api